### PR TITLE
[config] Improve how we use use SecureBackend

### PR DESCRIPTION
**Pull request by davidiw**
* Move default initialization to test
* Use SecureBackend::TryInt::<Box<DynStorage>> where possible
* Eliminate an unnecessary default impl

also fix a security bug where we inadvertently print the token to a log

### CODE REVIEW

File: config/src/config/safety_rules_config.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ impl Default for SafetyRulesConfig {
     fn default() -> Self {
         Self {
             backend: SecureBackend::InMemoryStorage,
-            service: SafetyRulesService::Local,
+            service: SafetyRulesService::Thread,
```
[the remark this answers is not in the document]
Comment (davidiw) [reply config/src/config/safety_rules_config.rs:19]: I should have called this out... but generator was setting it to Thread and therefore all our tests and networks were using Thread. My intent had always been that generator should not be changing values, the configs themselves should match what we deploy.

File: secure/storage/src/storage.rs [hunk 2]
```diff
@@ -1,11 +1,80 @@
...
 /// This is the Libra interface into secure storage. Any storage engine implementing this trait
 /// should support both key/value operations (e.g., get, set and create) and cryptographic key
 /// operations (e.g., generate_key, sign_message and rotate_key).
 pub trait Storage: KVStorage + CryptoStorage {}
 
 impl<T> Storage for T where T: KVStorage + CryptoStorage {}
+
+/// This is a hack that allows us to convert from SecureBackend into a useable T: Storage, since
+/// Box<dyn Storage> is not T: Storage.
+pub struct BoxStorage(pub Box<dyn Storage>);
```
[the remark this answers is not in the document]
Comment (davidiw) [reply secure/storage/src/storage.rs:19]: Honestly, don't want to think about it :P

I do know we will be adding more storage types in the future and having to maintain that list in multiple places isn't scalable. I think this is good for now and we can evaluate alternatives in the future. I am really not excited about trait objects.

[the remark this answers is not in the document]
Comment (JoshLind) [reply secure/storage/src/storage.rs:19]: Sure, okay. To be honest, I think it's super ugly, but if you're happy with leaving it as-is just to "move fast", then that's okay. Maybe add a TODO above it though (for once we get some time!)

**Comment by huitseeker** [comment]
I'm really sad I missed this. This pretty much precludes converting `Storage` to the new signing discipline (signing messages instead of `HashValues`) on top of the correct and unavoidable questions asked [here](https://github.com/libra/libra/pull/3690/files#r418734659).

**Comment by davidiw** [comment]
> I'm really sad I missed this. This pretty much precludes converting `Storage` to the new signing discipline (signing messages instead of `HashValues`) on top of the correct and unavoidable questions asked [here](https://github.com/libra/libra/pull/3690/files#r418734659).

Perhaps you could elaborate here.

**Comment by huitseeker** [comment]
See #4368.

**Pull request closed without merging**